src/process.rs

```diff
@@ -395,19 +437,19 @@ where
 ///
 /// `ProcessRef<T>` is different from a `Process` in the ability to handle messages of different
 /// types, as long as the traits `MessageHandler<M>` or `RequestHandler<R>` are implemented for T.
-#[derive(serde::Serialize, serde::Deserialize)]
-pub struct ProcessRef<T>
+#[derive(Serialize, Deserialize)]
+pub struct ProcessRef<T, S = Bincode>
 where
     T: ?Sized,
 {
-    process: Process<()>,
+    process: Process<(), S>,
     phantom: PhantomData<T>,
 }
 
-impl<T> ProcessRef<T> {
+impl<T, S> ProcessRef<T, S> {
```

Comment:
Does `S` need to bound to `ProcessRef`?

Reply:
I assume you mean just for this implementation?

If so, then the following code snippet requires the S generic for `ProcessRef::new`
https://github.com/lunatic-solutions/lunatic-rs/blob/b1d102cc72e0ee6882a01e8c50e195ba427505c5/src/process.rs#L202-L214

Reply:
I see, we need it on `ProcessRef` or we couldn't write `where  T: AbstractProcess<S>`.